Revert commit 966ca7e5d00a4b0bba5d6d9392303231e4e4e536 from upstream
(enforcing c++98), because it breaks builds with newer poppler.

Unbreak build: we don't provide library symlinks

Index: CMakeLists.txt
--- CMakeLists.txt.orig
+++ CMakeLists.txt
@@ -118,7 +118,11 @@ else()
   set(GDCM_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
 endif()
 
-if(BUILD_SHARED_LIBS)
+# OpenBSD doesn't have library symlinks
+if(CMAKE_SYSTEM_NAME MATCHES OpenBSD)
+  set(NAMELINK_ONLY)
+  set(NAMELINK_SKIP)
+elseif(${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION} GREATER 2.6)
   set(NAMELINK_ONLY NAMELINK_ONLY)
   set(NAMELINK_SKIP NAMELINK_SKIP)
 endif()
